samples/js-schoolAgent/src/attendanceAgent.ts

```diff
@@ -16,16 +16,19 @@
 
 import { ai } from './genkit.js';
 import { reportAbsence, reportTardy } from './tools.js';
+import { agentDescription, agentPrompt } from './util.js';
+
+const tools = [reportAbsence, reportTardy];
+const specialization = 'attendance';
 
 export const attendanceAgent = ai.definePrompt(
   {
-    name: 'attendanceAgent',
-    description:
-      'transfer to this agent when the user asks questions about attendance-related concerns like tardies or absences. do not mention that you are transferring, just do it',
-    tools: [reportAbsence, reportTardy],
+    name: `${specialization}Agent`,
+    description: agentDescription(specialization, tools),
+    tools,
   },
   ` {{ role "system"}}
-  You are Bell, a helpful attendance assistance agent for Sparkyville High School. A parent has been referred to you to handle an attendance-related concern. Use the tools available to you to assist the parent.
+${agentPrompt(specialization)}
```

Comment:
This makes it hard to see what this agent does at a glance. I think I little duplication is OK. It makes customization easier.